.github/workflows/build.yml

```diff
@@ -185,60 +191,54 @@ jobs:
         with:
           submodules: true
       - uses: DataDog/action-prebuildify/test@main
+        if: matrix.node >= inputs.minimum-major-version
 
   linux-test:
+    strategy:
+      matrix:
+        node: [12, 14, 16, 18, 20]
     needs: linux-x64
     runs-on: ubuntu-latest
-    name: linux-test
+    name: linux-test-${{ matrix.node }}
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: true
-      - uses: DataDog/action-prebuildify/node/12@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/14@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/16@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/18@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/20@main
       - uses: DataDog/action-prebuildify/test@main
+        if: matrix.node >= inputs.minimum-major-version
 
   macos-x64-test:
+    strategy:
+      matrix:
+        node: [12, 14, 16, 18, 20]
     needs: macos
     runs-on: macos-11
-    name: macos-x64-test
+    name: macos-x64-test-${{ matrix.node }}
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: true
-      - uses: DataDog/action-prebuildify/node/12@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/14@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/16@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/18@main
-      - uses: DataDog/action-prebuildify/test@main
-      - uses: DataDog/action-prebuildify/node/20@main
+      - uses: actions/setup-node@v2
```

Comment:
Let's add the condition on this too if it cannot be on the entire job, because this is by far the slowest step on macOS and Windows.